documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -0,0 +1,77 @@
+[id='file-stores_{context}']
+= File-Based Cache Stores
+
+File-based cache stores provide persistent storage on the local host filesystem where {brandname} is running.
+For clustered caches, file-based cache stores are unique to each {brandname} node.
+
+You should use file-based cache stores to persist data that overflows from memory when the data container exceeds size or time restrictions.
+
+[WARNING]
+====
+Never use filesystem-based cache stores on shared file systems, such as an NFS or Samba share, because they do not provide file locking capabilities and data corruption can occur.
+
+Additionally if you attempt to use transactional caches with shared file systems, unrecoverable failures can happen when writing to files during the commit phase.
+====
+
+[discrete]
+== Soft-Index File Stores
+
+[NOTE]
+====
+`SoftIndexFileStore` is the default implementation for file-based cache stores.
+====
+
+`SoftIndexFileStore` is a Java implementation that uses a variant of **B+
```

Comment:
I would say it is more important to say that soft index stores data in append only files which are then indexed using a B+ Tree that can be both stored on disk and in memory, with the latter being removed if necessary by the GC with soft references.

Once the append only file reaches its configured max size it will create a new file and start writing only to the new one.

Old data files will be compacted when their size reaches the compaction threshold which defaults to 50% removing the old files.

I am not sure how you want to word it exactly.